src/core/transfer_request.h

```diff
@@ -34,6 +38,19 @@ class nixlXferReqH {
         nixl_xfer_op_t     backendOp;
         nixl_status_t      status;
 
+        nixl_status_t
+        handleFailure(nixlAgentData &data, nixl_status_t status) {
+            NIXL_DEBUG << "Handling failure: " << nixlEnumStrings::statusStr(status);
+
+            if (status == NIXL_ERR_REMOTE_DISCONNECT) {
+                data.invalidateRemoteData(remoteAgent);
+            }
+
+            delete this;
```

Comment:
I'm afraid it could cause problems.
I suggest to keep `delete` in scope of `nixlAgent`, since the corresponding `new` operator is located there.
I understand that using raw pointers is a problem in itself. But I hope that putting `new` and `delete` in the same scope will make it easier to refactor in the future.

Reply:
ok, reverted